test/agentchat/test_conversable_agent.py

```diff
@@ -624,6 +634,161 @@ async def exec_sh(script: Annotated[str, "Valid shell script to execute."]):
         assert get_origin(user_proxy_1.function_map) == expected_function_map
 
 
+@pytest.mark.skipif(
+    skip or not sys.version.startswith("3.10"),
+    reason="do not run if openai is not installed or py!=3.10",
+)
+def test_function_restration_e2e_sync() -> None:
```

Comment:
added end to end integration test for both `sync` and `async` case

Reply:
```suggestion
def test_function_registration_e2e_sync() -> None:
```